tests/test_ec2_InstanceTags.py

```diff
@@ -0,0 +1,186 @@
+                    'op': 'in',
+                    'key': 'InstanceId',
+                    'value': [
+                        ec2_metadata_tags_disabled['aws_instance.metadata_tags.id'],
+                    ],
+                },
+                {'MetadataOptions.InstanceMetadataTags': 'disabled'},
+                {'not': [{'type': 'set-metadata-access'}]},
+            ],
+        },
+        session_factory=session_factory,
+        config={'region': aws_region},
+    )
+
+    resources = p.run()
+    test.assertEqual(len(resources), 2)
+
+    resource_ids = [i['InstanceId'] for i in resources]
+    test.assertIn(
+        ec2_metadata_tags_disabled['aws_instance.termination_protection.id'],
+        resource_ids)
+    test.assertIn(
+        ec2_metadata_tags_disabled['aws_instance.no_protection.id'],
+        resource_ids)
+
+    # set the api stop protection to false to allow terraform to handle the teardown
+    client = session_factory().client('ec2')
+    client.modify_instance_attribute(
+        InstanceId=ec2_metadata_tags_disabled['aws_instance.metadata_tags.id'],
+        DisableApiStop={'Value': False}
+    )
+
+
+class TestSetMetadata(BaseTest):
+
+    def test_set_metadata_server(self):
+        output = self.capture_logging('custodian.actions')
+        session_factory = self.replay_flight_data('test_ec2_set_md_access')
+        policy = self.load_policy({
+            'name': 'ec2-imds-access',
+            'resource': 'aws.ec2',
+            'actions': [
+                {'type': 'set-metadata-access',
+                 'tokens': 'required'},
+            ]},
+            session_factory=session_factory)
+        resources = policy.run()
+        if self.recording:
+            time.sleep(2)
+        results = session_factory().client('ec2').describe_instances(
+            InstanceIds=[r['InstanceId'] for r in resources])
+        self.assertJmes('[0].MetadataOptions.HttpTokens', resources, 'optional')
+        self.assertJmes(
+            'Reservations[].Instances[].MetadataOptions',
+            results,
+            [{'HttpEndpoint': 'enabled',
+              'HttpPutResponseHopLimit': 1,
+              'HttpTokens': 'required',
+              'InstanceMetadataTags': 'disabled',
+              'State': 'pending'},
+             {'HttpEndpoint': 'enabled',
+              'HttpPutResponseHopLimit': 1,
+              'HttpTokens': 'required',
+              'InstanceMetadataTags': 'disabled',
+              'State': 'applied'}])
+        self.assertEqual(len(resources), 2)
+        self.assertEqual(
+            output.getvalue(),
+            ('set-metadata-access implicitly filtered 1 of 2 resources '
+             'key:MetadataOptions.HttpTokens on optional\n'))
+
+
+class TestSetMetadataTags(BaseTest):
+
+    def test_set_metadata_server(self):
+        output = self.capture_logging('custodian.actions')
+        session_factory = self.replay_flight_data('test_ec2_set_md_access')
```

Comment:
this needs to be a separate directory, else its overwriting other tests.

Reply:
The file has been moved to tests/ec2_metadatatags/test_ec2_InstanceTags.py